models/threshold/Threshold.adoc

```diff
@@ -0,0 +1,83 @@
+= ThresholdHealth
+:toc: macro
+:toc-title:
+
+== Introduction
+
+Kiali provides a customize configuration for ThresholdAlert from requests in your environment.
+
+== Table of contents
+
+toc::[]
+
+== Configuration
+
+[source,yaml]
+----
+thresholds_health:
+  - namespace: "bookinfo"
+    threshold_checks:
+      - rule: ">20"
+        kind: "all"
+        label: "response_code"
+        filter:
+          reg: "views"
+          label_filter:
+            labels:
+              - app: "^details$"
+                version: "^v1$"
+            operation: "and"
+        expression: "400>x"
+        alert: "warning"
+----
+
+- *namespace*: Namespace to apply the ThresholdHealth
+- *theshold_checks*: List of Checks for that namespace
+  * *rule*: The expression to match.(In the sample the selected requests must be greater than the 20%)
+  * *kind*: Resources to apply. Options (`service`, `workload`, `app`, `all`/`""`)
+  * *label*: Label of request in Prometheus that we want to have the ThresholdHealth
+  * *filter*: Options for filter requests
+    ** *reg*: Regex that the `name` of the resource must match
+    ** *label_filter*: Filter by labels of the resource (See <<LabelFilter and PromFilter configuration>>)
+    ** *prom_filter*: Filter requests by prometheus label (See <<LabelFilter and PromFilter configuration>>)
+  * *expression*: The condition for requests for `label` (In example, requests where `response_code` is greater than `400`)
+  * *alert*: The alert that we want to show in the UI. (The message in UI will be [`alert`] `format message provided by kiali`. Options should be something like `INFO`, `WARNING` ....)
+
+
+=== Filter configuration
+
+We can filter the Threshold by:
+
+- Regex: Match the resource name by this regex
+- LabelFilter: Filter by labels of the resource
+- PromFilter: Filter by Prometheus Labels of each request
+
+
+==== LabelFilter and PromFilter configuration
```

Comment:
I'm must say I'm not very confident about that part. I understand the goal is to have some powerful matching capabilities on labels, but I've the feeling that writing complex matching rules exploded in yaml structures, with potentially several levels of nested objects, could turn into a bad experience. I'd like to hear others about it. Perhaps it would need further discussion / brainstorming .. Perhaps more research including about if there's some DSL parsers that already exist in the wild that we could reuse here ?

Reply:
My idea is have a powerful filtering here :) I think that the most complex filter will have 2 levels
AND
--> Labels with OR
--> Labels with OR

You can use a regex pattern for or values in the same label too so I didn't expect a huge yaml config for this. But yes we can can discuss about this and have a brainstorming session

Reply:
I'm in the middle of the review, but I share some of the concerns as well.

I also understand that this feature can be very open, but as it is, it could be seen like a proto-alerting-engine and that's not really the goal.

The idea is to have a first step to extract implicit hardcoded logic used to calculate health in a way the user can modify globally from the config.

But I haven't yet finished to review completely the PR I need to understand better the approach.